R/create_task_makefile.R

```diff
@@ -59,63 +59,48 @@
 #'   task_plan, makefile=file.path(tempdir(), 'states.yml'),
 #'   file_extensions=c('ind'), packages='mda.streams')
 #' cat(readLines(task_makefile), sep='\n')
+#' @importFrom methods getPackageName
+#' @importFrom utils packageVersion
 create_task_makefile <- function(
   task_plan, makefile,
-  include=c(), packages=c(), sources=c(), file_extensions=c('ind'),
-  ind_complete=NA,
-  ind_dir=attr(task_plan, 'ind_dir'),
-  ind_ext=getOption("scipiper.ind_ext"),
-  template_file=system.file('extdata/task_makefile.mustache', package='scipiper')) {
+  include=c(), packages='scipiper', sources=c(), file_extensions=c('ind'),
+  template_file=system.file('extdata/task_makefile.mustache', package='scipiper'),
+  target_name, 
+  finalize_fun = 'combine_to_ind', 
+  as_promises = TRUE) {
   
   # prepare the overall job task: list every step of every job as a dependency.
   # first mutate the makefile file name into an object name to use as the
   # default/overall job target. this should be an acceptable target name (not
   # conflicting with other targets) and allows the calling remake file to use
   # ind_file as a target (even though this target is the one responsible for
   # actually writing to ind_file)
-  job_target <- tools::file_path_sans_ext(basename(makefile))
+  if (is.null(makefile) & missing(target_name)){
+    stop('target_name must be specified when makefile is NULL')
+  }
+  # default is to use the makefile naming to assign job name
+  # when makefile isn't used (string output), we use the target_name to create default job name
+  job_name <- ifelse(is.null(makefile), 
+                     paste0(tools::file_path_sans_ext(basename(target_name)), '_all'), 
```

Comment:
What if you want to output the makefile as string but you don't want any combiners (or have multiple combiners)? Could the job_name be something bland like `all_tasks`?

Reply:
Good idea, will update to that